src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -273,12 +273,6 @@ public String buildFullName(BuilderContext context) {
         }
     }
 
-    private static final ThreadLocal<List<Field>> FIELD_LIST = new ThreadLocal<List<Field>>() {
```

Comment:
I wonder if this should be a different change and we should look at the case where some docs have 1000 fields. Are we sure the auto-growth of the field list is not worth caching?